vaibhav/cart/shopping_cart.rb

```diff
@@ -0,0 +1,128 @@
+#shopping cart 
+
+class Shopping_cart
+
+    def initialize
+        @ary = Array.new{{}}
+        @cart = Array.new{{}}
+    end
+
+    def Add_Products(ary)
+        print 'Enter product name: '
+        productName = gets
+        print 'enter product value: '
+        productValue = gets
+        print 'Enter product quantity: '
+        productQuantity = gets
+        @ary.push([productName, productValue, productQuantity])
+    end
+
+    def AddTo_cart(cart, ary)
+
+        print 'Enter the product name: '
+        prdName = gets
+        @ary = [["apple",120,5],["orange",150,10]]
+        @cart = []
+      
+        abc = @ary.find { |x| x[0] == prdName }
+        if abc
+          print 'Enter qunatity of ', prdName, ' '
+          prdQuan = gets
+          @cart.push([prdName, abc[1], prdQuan])
+        else
+          return 'enter a valid name'
+        end
+        return @cart
+    end
+
+    def DisplayAll(ary)
+        puts "\ndisplaying all the products"
+        puts ' '
+        puts 'Name  Cost  Quantity'
+        @ary = [["apple",120,5],["orange",150,10]]
+        @ary.each do |r|
+          puts r.each { |p| p }. join(' ')
+        end
+        return @ary
+        puts ' '
+    end
+      
+    def Display_cart(cart)
+        puts "\ndisplaying your cart"
+        puts ' '
+        puts 'Name  Cost  Quantity'
+
+        @cart = [["apple",120,5]]
+
+        @cart.each do |r|
+          puts r.each { |p| p }. join(' ')
+        end
+        puts ' '
+        return @cart
+    end
+
+    def DeleteFrom_cart(cart)
+        puts 'which cart item u want to delete'
+        prdName = gets
+        @cart = [["apple",120,5],["orange",150,10]]
+
+        isDelete = @cart.find { |x| x[0] == prdName }
+        if isDelete
```

Comment:
you can remove this if condition check here because delete_if deletes the element only if it is present